pkg/model/api/v1/variable/datasource.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package variable
+
+import (
+	"fmt"
+)
+
+type DatasourceSpec struct {
+	Display *Display `json:"display,omitempty" yaml:"display,omitempty"`
+	// Value from the list to be selected by default.
+	DefaultValue *DefaultValue `json:"defaultValue,omitempty" yaml:"defaultValue,omitempty"`
+	// CapturingRegexp is the regexp used to catch and filter the list of datasources.
+	// If empty, then nothing is filtered. That's the equivalent of setting CapturingRegexp with (.*)
+	CapturingRegexp string `json:"capturingRegexp,omitempty" yaml:"capturingRegexp,omitempty"`
+	// Sort method to apply when rendering the list of datasources
+	Sort *Sort `json:"sort,omitempty" yaml:"sort,omitempty"`
+	// Type is the type (= kind) of datasources
+	Type string `json:"type" yaml:"type"`
```

Comment:
why not using Kind in that case ? We never used the word `Type` in the datamodel. I don't think we should start here to use it

Reply:
I had a feeling that this would look weird:

```
kind: ...
spec: {
  kind: ...
  ...
}
```

but I'm ok to change it.